ddsrouter_event/include/ddsrouter_event/wait/CounterWaitHandler.hpp

```diff
@@ -23,18 +23,35 @@
 #include <condition_variable>
 #include <mutex>
 
+#include <ddsrouter_event/library/library_dll.h>
 #include <ddsrouter_event/wait/WaitHandler.hpp>
 
 namespace eprosima {
 namespace ddsrouter {
 namespace event {
 
-using CounterType = int32_t;
+using CounterType = unsigned int;
 
 /**
- * @brief This WaitHandler allows to wait for an int value
+ * @brief This WaitHandler class wait threads for a counter value.
  *
- * This works as an int value that allows to wait for a specific value or threshold.
+ * This class could be understood as: the difference between internal value - threshold is the number of threads
+ * that could be awaken.
+ *
+ * This class holds an internal value that can be incremented or decremented one by one.
+ * It also has a constant threshold value.
+ * While the internal value overcome the threshold (higher), wait condition would be true for one of the threads.
+ * Once a thread has finished waiting, it decrements the internal value in 1.
+ *
+ * @note This is specialization of \c WaitHandler that is improved for be more efficient then \c IntWaitHandler
+ * as it only notify threads when the actually need to wake up, and only one thread at a time.
+ * Very useful for consumer wait handlers.
+ *
+ * @note This class is thread safe and do work properly. Side cases could be:
+ * 1. Value is higher than 1+threshold -> the waiting thread that is awaken will
+ * decrease value by 1, and then next thread will be notified (one by one)
+ * 2. Value is higher than threshold before thread arrive to wait -> predicate is checked
+ * at the instantiation time, so it does not need a notify.
  */
 class CounterWaitHandler : protected WaitHandler<CounterType>
```

Comment:
Will you be adding tests for this class? I guess it's not really worth it, as its functionality is tested through its children.

Reply:
I did not plan to. It would be nice though.